lib/luahid.c

```diff
@@ -50,6 +47,35 @@ static const lunatik_class_t luahid_class = {
 	.sleep = true,
 };
 
+static const struct hid_device_id *luahid_setidtable(lua_State *L, int idx)
+{
+	size_t len = luaL_len(L, idx);
+	struct hid_device_id *user_table = lunatik_checkalloc(L, sizeof(struct hid_device_id) * (len + 1));
+
+	struct hid_device_id *cur_id = user_table;
+	for (size_t i = 0; i < len; i++, cur_id++) {
+		if (lua_geti(L, idx, i + 1) != LUA_TTABLE) { /* table entry */
+			lua_pop(L, 1); /* table entry */
+			lunatik_free(user_table);
+			user_table = NULL;
+			goto out;
+		}
+
+		lunatik_optinteger(L, -1, cur_id, bus, HID_BUS_ANY);
```

Comment:
You should export this enum to Lua as well, right? If you don't plan to do it in this PR, please, create an issue